lib/index.js

```diff
@@ -455,26 +445,28 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
       }
       var crossDomainId = null;
       var fromDomain = null;
+      var currentTimeMillis = (new Date()).getTime();
+      var timestamp = null;
       if (res) {
         crossDomainId = res.id;
         fromDomain = res.domain;
+        timestamp = res.timestamp || currentTimeMillis;
```

Comment:
Is it worth it to be explicitly here, eg.
```timestamp = 'timestamp' in res ? res.timestamp : currentTimeMillis```
to avoid bugs if when `res.timestamp` is set to a sentinel value such as `0`? Even if `res.timestamp` is set to `0` accidentally, I think it'd still be easier to spot the bug.